### ISSUE TITLE
Add s3 domain names with region so cloudfront origins outside of us-east-1 won't cause redirects immediately after setup

### ISSUE THREAD

**Issue by kitsunde** [terraform-providers/terraform-provider-aws#4540]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Buckets created outside of us-east-1 will redirect to s3 when set as origins for cloudfront until DNS propagation has happened *unless* using the regional domain name. We need to set:

```hcl
${aws_s3_bucket.default.bucket}.s3-${aws_s3_bucket.default.region}.amazonaws.com
```

It would be nice if we could get a `regional_domain_name` or maybe `canonical_domain_name` from the `aws_s3_bucket` resource, and even make it recommended for origin since it's rather surprising especially if the redirects get cached.

### New or Affected Resource(s)`

* aws_s3_bucket

### Potential Terraform Configuration

This

```hcl
  origin {
    domain_name = "${aws_s3_bucket.default.domain_name}"
    origin_id = "s3-${var.environment}"
    s3_origin_config {
      origin_access_identity = "${aws_cloudfront_origin_access_identity.default.cloudfront_access_identity_path}"
    }
  }
```

Needs to be rewritten as:

```hcl
  origin {
    domain_name = "${aws_s3_bucket.default.bucket}.s3-${aws_s3_bucket.default.region}.amazonaws.com"
    origin_id = "s3-${var.environment}"
    s3_origin_config {
      origin_access_identity = "${aws_cloudfront_origin_access_identity.default.cloudfront_access_identity_path}"
    }
  }
```

### References

* https://forums.aws.amazon.com/thread.jspa?threadID=216814

**Comment by saravanan30erd** [comment]
@bill-rich I would like to work on this issue.

**Comment by saravanan30erd** [comment]
PR submitted: #4556

**Comment by bflad** [comment]
The `bucket_regional_domain_name` attribute will be available in the `aws_s3_bucket` resource in version 1.21.0 of the AWS provider, releasing later today.

**Comment by bflad** [comment]
This has been released in [version 1.21.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1210-may-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
